pkg/apis/migration/v1alpha1/migcluster_types.go

```diff
@@ -107,16 +113,45 @@ func (m *MigCluster) GetServiceAccountSecret(client k8sclient.Client) (*kapi.Sec
 
 // GetClient get a local or remote client using a MigCluster and an existing client
 func (m *MigCluster) GetClient(c k8sclient.Client) (compat.Client, error) {
+	// Building a compat client requires both restConfig and a k8s client
+	var rClient *k8sclient.Client
+
+	// Retrieve cached client if it exists
+	if m.Spec.IsHostCluster {
+		rClient = &c
+	} else {
+		rwm := remote.GetWatchMap()
+		remoteCluster := rwm.Get(types.NamespacedName{Namespace: m.Namespace, Name: m.Name})
+		if remoteCluster != nil {
+			cachedClient := remoteCluster.RemoteManager.GetClient()
```

Comment:
correct me if I am wrong but `remoteCluster.RemoteManager.GetClient()` will not get us a cached client. Probably need to do a `remoteCluster.RemoteManager.GetCache()`

Reply:
Need to look into this, thanks for reviewing

Reply:
After discussion we determined this will get us a delegatingClient which will use cache for read.